x-pack/libbeat/autodiscover/providers/aws/elb/fetch.go

```diff
@@ -175,14 +184,23 @@ func (p *fetchRequest) dispatch(fn func()) {
 func (p *fetchRequest) fetchListeners(lb elasticloadbalancingv2.LoadBalancer) {
 	listenReq := p.client.DescribeListenersRequest(&elasticloadbalancingv2.DescribeListenersInput{LoadBalancerArn: lb.LoadBalancerArn})
 	listen := elasticloadbalancingv2.NewDescribeListenersPaginator(listenReq)
-	for listen.Next(context.TODO()) && p.running.Load() {
-		for _, listener := range listen.CurrentPage().Listeners {
-			p.recordGoodResult(&lb, &listener)
-		}
-	}
+
 	if listen.Err() != nil {
 		p.recordErrResult(listen.Err())
 	}
+
+Outer:
+	for listen.Next(p.context) {
+		select {
+		case <-p.context.Done():
+			break Outer
```

Comment:
Here we could also `return`. In any case, maybe `listen.Next` also returns `false` the passed context is done?